### DISCUSSION TITLE
.env or publicRuntime inside composables?

### DISCUSSION THREAD

**Discussion by LvL090** [nuxt/nuxt#16485] [Questions]
Hi all! 

I need to pass .env variables inside a composable.
I want to simply use Supabase within a composable but i can't find any way.
I have created the .env with the password and the URL and also configured the public route in the nuxt.config

`   import { createClient } from '@supabase/supabase-js'

const SUPABASE_URL = 'https://___'      <-- REPLACE STRING with .env
const SUPABASE_KEY = 'ey*****___'                        <-- REPLACE STRING with .env

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY)

export default function useSupabase () {
  return { supabase }
}     `

**Comment by TheAlexLichter** [comment]
You can use [`useRuntimeConfig`](https://v3.nuxtjs.org/api/composables/use-runtime-config#useruntimeconfig).

**Comment by wobsoriano** [comment]
FYI there’s a supabase module

https://supabase.nuxtjs.org/
